testdata/tests/cases/compiler/assertionWithNoArgument.ts

```diff
@@ -0,0 +1,8 @@
+// @strict: true
+// @noemit: true
+
+export function assertWeird(value?: string): asserts value {
+}
+
+assertWeird();
+assertWeird("hello");

```

Comment:
Might be worth having another test that swaps the order?

Reply:
Nah, isn't really relevant. See my comment below.